File: OriginalArtifact-SWE/AppointmentService/src/main/AppointmentService.java
package main;
import java.util.*;

public class AppointmentService {
	
	public HashMap<String, Appointment> appointments = new HashMap<>();
	
	public AppointmentService() {}
	
	public void addAppointment(Appointment appointment) {
		if (appointment == null) {
			throw new IllegalArgumentException("Appointment is null");
		}
		appointments.put(appointment.getAppointmentId(), appointment);
	}
	
	public void deleteAppointment(String appointmentId) {
		if (appointmentId == null || appointmentId == "" || appointmentId.length() > 10) {
			throw new IllegalArgumentException("Invalid appointment ID");
		}
		appointments.remove(appointmentId);
	}
}

File: ArtifactEnhancement-SWE/AppointmentService.cpp
#include "AppointmentService.hpp"

/*
        Implementations of function declarations provided by AppointmentService.h
*/

// Constructor and destructor
AppointmentService::AppointmentService() {}
AppointmentService::~AppointmentService() {}

// Singleton instance
AppointmentService& AppointmentService::getInstance() {
    static AppointmentService instance;
    return instance;
}

// Add appointment
void AppointmentService::addAppointment(const Appointment& appointment) {
    appointments[appointment.getId()] = appointment;
}

// Update task
void AppointmentService::updateAppointment(const std::string& appointmentId, const Appointment& appointment) {
    appointments[appointmentId] = appointment;
}

// Delete task
void AppointmentService::deleteAppointment(const std::string& taskId) {
    appointments.erase(taskId);
}

// Number of task
int AppointmentService::numberOfAppointments() const {
    return appointments.size();
}